source/FreeRTOS_IP.c

```diff
@@ -925,21 +949,22 @@ BaseType_t FreeRTOS_IPInit_Multi( void )
 
     pxFirstEndPoint = FreeRTOS_FirstEndPoint( NULL );
 
-    if( ENDPOINT_IS_IPv6( pxFirstEndPoint ) )
-    {
-        for( ;
-             pxFirstEndPoint != NULL;
-             pxFirstEndPoint = FreeRTOS_NextEndPoint( NULL, pxFirstEndPoint ) )
+    #if ( ipconfigUSE_IPv6 != 0 )
+        if( ENDPOINT_IS_IPv6( pxFirstEndPoint ) )
         {
-            if( ENDPOINT_IS_IPv4( pxFirstEndPoint ) )
+            for( ;
+                 pxFirstEndPoint != NULL;
+                 pxFirstEndPoint = FreeRTOS_NextEndPoint( NULL, pxFirstEndPoint ) )
             {
-                break;
+                if( ENDPOINT_IS_IPv4( pxFirstEndPoint ) )
+                {
+                    break;
+                }
             }
         }
-    }
-
-    /* At least one IPv4 end-point must be defined. */
-    configASSERT( pxFirstEndPoint != NULL );
+    #else /* if ( ipconfigUSE_IPv6 != 0 ) */
+        ( void ) pxFirstEndPoint;
```

Comment:
Why is else needed? We can remove it.

Reply:
`pxFirstEndPoint` is un used if `ipconfigUSE_IPv6 == 0`